modules/libpng/1.6.40/patches/add_build_file.patch

```diff
@@ -0,0 +1,62 @@
+--- BUILD.bazel
++++ BUILD.bazel
+@@ -0,0 +1,59 @@
++# Copied from https://github.com/google-deepmind/lab2d/blob/09474439959534b1de3fce3d4a4ed35d565ea230/bazel/png.BUILD 
++# with minor modifications available under Apache 2.0 License
++# SPDX-License-Identifier: Apache-2.0
++
++# Description:
++#   libpng is the official PNG reference library.
++
++genrule(
++    name = "pnglibconf",
++    srcs = ["scripts/pnglibconf.h.prebuilt"],
++    outs = ["pnglibconf.h"],
++    cmd = "cp $(location scripts/pnglibconf.h.prebuilt) $(location pnglibconf.h)",
++)
++
++cc_library(
++    name = "libpng",
++    srcs = [
++        "png.c",
++        "pngdebug.h",
++        "pngerror.c",
++        "pngget.c",
++        "pnginfo.h",
++        "pnglibconf.h",
++        "pngmem.c",
++        "pngpread.c",
++        "pngpriv.h",
++        "pngread.c",
++        "pngrio.c",
++        "pngrtran.c",
++        "pngrutil.c",
++        "pngset.c",
++        "pngstruct.h",
++        "pngtrans.c",
++        "pngwio.c",
++        "pngwrite.c",
++        "pngwtran.c",
++        "pngwutil.c",
++    ] + select({
++        "@platforms//cpu:arm64": [
++            "arm/arm_init.c",
++            "arm/filter_neon_intrinsics.c",
++            "arm/palette_neon_intrinsics.c",
++        ],
++        "@apple_support//configs:darwin_arm64": [
```

Comment:
If this branch is identical with the previous one and more specific, is it needed?